sbor/src/schema/schema_comparison.rs

```diff
@@ -0,0 +1,2018 @@
+                }
+            }
+            TypeKind::Enum {
+                variants: base_variants,
+            } => {
+                let TypeKind::Enum {
+                    variants: compared_variants,
+                } = compared_type_kind
+                else {
+                    return result.with_mismatch_error(base_type_kind, compared_type_kind);
+                };
+
+                let base_variants_missing_in_compared: IndexSet<_> = base_variants
+                    .keys()
+                    .filter(|base_variant_id| !compared_variants.contains_key(*base_variant_id))
+                    .cloned()
+                    .collect();
+                let compared_variants_missing_in_base: IndexSet<_> = compared_variants
+                    .keys()
+                    .filter(|compared_variant_id| !base_variants.contains_key(*compared_variant_id))
+                    .cloned()
+                    .collect();
+
+                if base_variants_missing_in_compared.len() > 0
+                    || (compared_variants_missing_in_base.len() > 0
+                        && !settings.allow_new_enum_variants)
+                {
+                    result.add_error(SchemaComparisonErrorDetail::EnumSupportedVariantsMismatch {
+                        base_variants_missing_in_compared,
+                        compared_variants_missing_in_base,
+                    });
+                }
+
+                for (discriminator, base_field_type_ids) in base_variants.iter() {
+                    let Some(compared_field_type_ids) = compared_variants.get(discriminator) else {
+                        // We have already output a EnumSupportedVariantsMismatch error above for this.
+                        // But let's continue to see if we can match / compare further variants structurally,
+                        // to get as many errors as we can.
+                        continue;
+                    };
+                    let discriminator = *discriminator;
+
+                    if base_field_type_ids.len() != compared_field_type_ids.len() {
+                        result.add_error(
+                            SchemaComparisonErrorDetail::EnumVariantFieldCountMismatch {
+                                variant_discriminator: discriminator,
+                                base_field_count: base_field_type_ids.len(),
+                                compared_field_count: compared_field_type_ids.len(),
+                            },
+                        );
+                    } else {
+                        let paired_child_ids = base_field_type_ids
+                            .iter()
+                            .zip(compared_field_type_ids.iter())
+                            .enumerate();
+                        for (field_index, (base_child_type_id, compared_child_type_id)) in
+                            paired_child_ids
+                        {
+                            result.add_child_to_check(
```

Comment:
We don't visit elements in `compared_variants_missing_in_base`? This may cause completeness check to fail due to children types of these types?

Reply:
Well `compared_variants_missing_in_base` have nothing to compare against, so they're trivially okay (if they're allowed in the first place).

The completeness checks are purposefully done separately to the type comparison checks for this reason -- even if we don't type check a base type against a compared type for some reason (e.g. a reason like this; or a type comparison failure further up its ancestry), we still want to check if it's reachable from a root type.